File: ride.cpp
/*
ID: totheso1
LANG: C++14
TASK: ride
*/

#include <fstream>
#include <iterator>
#include <numeric>
#include <string>

using namespace std;

static ifstream fin("ride.in");
static ofstream fout("ride.out");

static int convert(const string &name) {
    // if C++17: std::transform_reduce
    int ret = 1;
    for (const char c : name) {
        ret *= c - 64;
    }
    return ret % 47;
}

int main() {
    istream_iterator<string> lines_iter(fin);
    fout << (convert(*lines_iter++) == convert(*lines_iter) ? "GO" : "STAY")
         << endl;
}

File: ride.py
"""
ID: totheso1
LANG: PYTHON3
TASK: ride
"""

import functools
import itertools
import operator

fin = open("ride.in", "r")
fout = open("ride.out", "w")


def fprint(*args, **kwargs) -> None:
    print(*args, file=fout, **kwargs)


# Code start
def convert(name: str) -> int:
    return (
        functools.reduce(
            operator.mul,
            map(lambda c: ord(c) - 64, itertools.islice(name, len(name) - 1)),
        )
        % 47
    )


lines = fin.readlines()
fprint("GO" if convert(lines[0]) == convert(lines[1]) else "STAY")
# Code end
